test_app/tests/rbac/test_ansible_id_alias_filter.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+from django.contrib.contenttypes.models import ContentType
+from django.utils.http import urlencode
+
+from ansible_base.lib.utils.response import get_relative_url
+from ansible_base.resource_registry.models import Resource
+
+
+class TestAnsibleIdAliasFilterBackend:
+
+    @pytest.mark.django_db
+    def test_filter_user_ansible_id(self, admin_api_client, org_inv_rd, rando, organization):
+        '''
+        Test filtering RoleUserAssignment by user_ansible_id and object_ansible_id.
+        '''
+        user_resource = Resource.objects.get(object_id=rando.pk, content_type=ContentType.objects.get_for_model(rando).pk)
+        organization_resource = Resource.objects.get(object_id=organization.pk, content_type=ContentType.objects.get_for_model(organization).pk)
+        url = get_relative_url('roleuserassignment-list')
+        data = dict(role_definition=org_inv_rd.id, content_type='shared.organization', user_ansible_id=user_resource.ansible_id, object_id=organization.id)
+        response = admin_api_client.post(url, data=data, format="json")
+        assert response.status_code == 201, response.data
```

Comment:
I think you should check that the total number of user_role_assignments is >1, and if not, assign another permission to any random user for any random object.

Otherwise there's a risk that this test doesn't cover the null case in any way. It's returning the desired object, but it's not clear that it doesn't return undesired objects.

Test is good, I think it's overkill to add null-case testing as a separate test. But doing a two-liner additional assignment would cover the bases fine.

Reply:
add some logic to ensure we have at least 2 assignments, and that filtering returns the expected number of items